spec/models/labware_metadata_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LabwareMetadata do
+  include FeatureHelpers
+
+  describe 'with api' do
+    let(:user_uuid)           { SecureRandom.uuid }
+    let(:user)                { json :v1_user, uuid: user_uuid }
+    let(:plate_uuid)          { SecureRandom.uuid }
+    let(:plate)               { json :stock_plate, uuid: plate_uuid }
+    let(:plate_with_metadata) { json :stock_plate_with_metadata, uuid: plate_uuid }
+
+    let(:tube_uuid)          { SecureRandom.uuid }
+    let(:tube)               { json :stock_tube, uuid: tube_uuid }
+    let(:tube_with_metadata) { json :stock_tube_with_metadata, uuid: tube_uuid }
+
+    has_a_working_api
+
+    it 'raises an exception if the barcode is invalid' do
+      stub_asset_search(456, nil)
+      expect do
+        LabwareMetadata.new(api: api, barcode: 456, user: user_uuid)
+      end.to raise_error(Sequencescape::Api::ResourceNotFound)
+    end
+
+    context 'plates' do
```

Comment:
There's a lot of duplication in these specs. Might be cleaner to set-up the two contexts and use some shared examples.